### ISSUE TITLE
In insufficient capacity case, distinguish between system capacity and user-defined IG capacity

### ISSUE THREAD

**Issue by AlanCoding** [ansible/awx#11316]
##### ISSUE TYPE
 - Feature Idea

##### SUMMARY
Why will this job not start?

![Screenshot from 2021-11-08 10-24-27](https://user-images.githubusercontent.com/1385596/140769643-b844c33f-9adc-4079-8ee5-4ef0b145d351.png)

from:
https://github.com/ansible/awx/blob/62e9e7ea80d9b931e33432a3d393bf29be886c2d/awx/main/scheduler/task_manager.py#L554

There's not capacity available, but capacity _of what_? The system has plenty of capacity.

If the user navigates from the job details to the job template details, they can find:

![Screenshot from 2021-11-08 10-27-41](https://user-images.githubusercontent.com/1385596/140769902-d71a1fc2-8a21-4d5f-a7bc-2c4b2c7906fe.png)

In this case, the root cause is that the `awx_2` instance group _has no instances in it_.

Custom instance groups could be defined on:

 - the job template
 - the project
 - the organization

So the steps above are not sufficient to diagnose the problem. In normal circumstances, the diagnosis process would be to check for custom instance groups on the JT, navigate to the project and check that, and then navigate to the organization and check that too. This is absurd.

----

This issue proposes that we should give some heads-up to the user if a job that won't start due to capacity is using custom instance groups.

If the job has no custom instance groups defined, then it should be waiting for the `default` instance group to have capacity. In that case, this is a problem of overall _system_ capacity.

In short, we should distinguish capacity issues due to _system_ capacity versus custom instance group configuration.

**Comment by AlanCoding** [comment]
Looking at some more examples of this, I see some other directions we can go in.

If a job fits the described case, then we have we have:

 - some job being processed by the task manager
 - a list of preferred instance groups
 - each instance group in that list has 0 instances with non-zero capacity

Why should this job remain in "pending"? Pending what? Pending the user fixing their cluster configuration?

I think there's a decent argument that the job should not remain in an active state, and it should be marked as error right away.

Some people might disagree with that. I would be willing to back off the final criteria that we error jobs that have no _enabled_ instances in their preference list. This will never be fixed except by re-running the installer. If we used that weaker criteria, I think we should still have some distinct messaging for the case of enabled instances with zero capacity.
